### PR TITLE
range-based for

### PR DESCRIPTION
**Pull request by tjhei**

### CODE REVIEW

File: include/aspect/boundary_composition/interface.h [hunk 1]
```diff
@@ -346,15 +346,13 @@ namespace aspect
...
-        if (Plugins::plugin_type_matches<BoundaryCompositionType>(*(*p)))
-          return Plugins::get_plugin_as_type<BoundaryCompositionType>(*(*p));
+      for (const auto &p : boundary_composition_objects)
+        if (Plugins::plugin_type_matches<BoundaryCompositionType>(*p))
+          return Plugins::get_plugin_as_type<BoundaryCompositionType>(*p);
 
       // We will never get here, because we had the Assert above. Just to avoid warnings.
-      return Plugins::get_plugin_as_type<BoundaryCompositionType>(*(*boundary_composition_model));
+      typename std::vector<std::unique_ptr<Interface<dim>>>::const_iterator iterator;
+      return Plugins::get_plugin_as_type<BoundaryCompositionType>(*(*iterator));
```
Comment (bangerth) [inline include/aspect/boundary_composition/interface.h:355]: This is not much better than doing `return *nullptr;`...

**Review by bangerth — APPROVED** [review]

File: include/aspect/boundary_composition/interface.h (hunk 1 above)
Comment (tjhei) [reply include/aspect/boundary_composition/interface.h:355]: except you can not dereference a nullptr, I think.

Comment (bangerth) [reply include/aspect/boundary_composition/interface.h:355]: You'll get about the same result as dereferencing an uninitialized iterator ;-)

Comment (bangerth) [reply include/aspect/boundary_composition/interface.h:355]: The difference is that the compiler will warn you about dereferencing the `nullptr`.

**Pull request merged**